.githooks/commit-msg

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+
+NAME=$(git config user.name)
+EMAIL=$(git config user.email)
+AUTO_SIGN=$(git config user.auto-signoff)
+SELF_SIGN=$(grep -qi "^Signed-off-by: $NAME <$EMAIL>" "$1")
+
+if [ -z "$NAME" ]; then
+    echo "empty git config user.name"
+    exit 1
+fi
+
+if [ -z "$EMAIL" ]; then
+    echo "empty git config user.email"
+    exit 1
+fi
+
+# auto sign-off if configured
+if [ "$AUTO_SIGN" = true ]; then
+    git interpret-trailers --if-exists doNothing --trailer \
+    "Signed-off-by: $NAME <$EMAIL>" \
+    --in-place "$1"
+fi
```

Comment:
Shouldn't auto signoff mean that the `SELF_SIGN` check will be true? Or are these checks before the auto signoff is applied?

Reply:
`SELF_SIGN` means that the user has added the `-s` flag manually whereas `AUTO_SIGN` is a git config that is set using `config user.auto-signoff true`